File: cat2/lineCount.swift
import Foundation

#if os(Linux)
import Glibc
#endif

let argc = CommandLine.arguments.count
let argv = CommandLine.arguments

for file in argv[1..<argc] {
	var f: UnsafeMutablePointer<FILE>?
	var c: Int32
	var lineNum: UInt32 = 0
	let  asciiReturn: UInt32 = 10

	f = fopen(file, "r")
	if f == nil {
		perror(file)
		exit(1)
	}
	while true {
		c = fgetc(f)
		if c == EOF { break }
		if c == asciiReturn { lineNum += 1 }
	}
	print("\(file): \(lineNum)")
}

File: cat2/lineCount.c
#include <stdio.h>
#include <stdlib.h>


int main(int argc, char *argv[]) {
	int i;

	for (i = 1; i < argc; i++){
		FILE *f;
		int c;
		int lineNum = 0;
		char clineNum[128];

		f = fopen(argv[i], "r");
		if (!f) {
			perror(argv[i]);
			exit(1);
		}
		while ((c = fgetc(f)) != EOF) {
			if (c == '\n') lineNum++;
		}
		if (printf("%d\n", lineNum) < 0) exit(1);
		fclose(f);
	}
	exit(0);
}
